apps/dav/lib/CardDAV/CardDavBackend.php

```diff
@@ -1182,6 +1183,13 @@ private function searchByAddressBookIds(array $addressBookIds,
 			$query2->setFirstResult($options['offset']);
 		}
 
+		if(isset($options['person'])){
+			if('' !== $pattern){
+				$query2->orWhere($query2->expr()->ilike('cp.value', $query2->createNamedParameter('%' . $this->db->escapeLikeParameter($options['person']) . '%')));
```

Comment:
```suggestion
				$query2->andWhere($query2->expr()->ilike('cp.value', $query2->createNamedParameter('%' . $this->db->escapeLikeParameter($options['person']) . '%')));
```

to fix https://github.com/nextcloud/server/pull/41457#issuecomment-1810694416

Reply:
With this change the feature works as expected for me. If there is a regular contact with the same name as the filtered user, I will see the contact. If there is no match, I see nothing :+1: